File: chazm-model/src/main/java/io/github/runtimemodels/chazm/entity/CharacteristicEntity.java
package io.github.runtimemodels.chazm.entity;

import com.google.inject.assistedinject.Assisted;
import runtimemodels.chazm.api.entity.Characteristic;
import runtimemodels.chazm.api.id.UniqueId;

import javax.inject.Inject;
import javax.validation.constraints.NotNull;

class CharacteristicEntity extends AbstractEntity<Characteristic> implements Characteristic {

    @Inject
    CharacteristicEntity(@NotNull @Assisted final UniqueId<Characteristic> id) {
        super(id);
    }

    @Override
    public boolean equals(final Object object) {
        if (object instanceof Characteristic) {
            final Characteristic characteristic = (Characteristic) object;
            return super.equals(characteristic);
        }
        return false;
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }

    @Override
    public String toString() {
        return super.toString();
    }

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/entity/CharacteristicEntity.kt
package runtimemodels.chazm.model.entity

import com.google.inject.assistedinject.Assisted
import runtimemodels.chazm.api.entity.Characteristic
import runtimemodels.chazm.api.id.CharacteristicId
import javax.inject.Inject

internal data class CharacteristicEntity @Inject constructor(
    @Assisted override val id: CharacteristicId
) : Characteristic
